squad/core/tasks/__init__.py

```diff
@@ -86,18 +86,22 @@ def __validate_metrics(self, metrics_file):
         if type(metrics) != dict:
             raise exceptions.InvalidMetricsData.type(metrics)
 
-        for key, value in metrics.items():
-            if type(value) is str:
-                try:
-                    value = float(value)
-                except ValueError:
-                    raise exceptions.InvalidMetricsData.value(value)
-            if type(value) not in [int, float, list]:
-                raise exceptions.InvalidMetricsData.value(value)
-            if type(value) is list:
-                for item in value:
-                    if type(item) not in [int, float]:
+        for metric, value_dict in metrics.items():
+            for key, value in value_dict.items():
```

Comment:
I think you don't need this loop given there's just "unit" and "value" keys in "value_dict".

Reply:
@chaws , yup I have changed this and made the validation of metrics and the parsing handle both, the legacy metric with no unit and the new one. Also added the test you for this.